### PR TITLE
Removing disabled simulated device to stop sending events to it and a…

### PR DESCRIPTION
**Pull request by kapil-ms**
…dding enabled simulated device to start sending events to it

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@kapil-ms__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (kaagarw). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs [hunk 1]
```diff
@@ -349,6 +349,41 @@ await _virtualDeviceStorage.AddOrUpdateDeviceAsync(new InitialDeviceConfig()
...
+                        {
+                            DeviceId = repositoryDevice.DeviceProperties.DeviceID,
+                            HostName = _configProvider.GetConfigurationSettingValue("iotHub.HostName"),
+                            Key = securityKeys.PrimaryKey
+                        });
+                    }
+                    catch (Exception ex)
+                    {
+                        //if we fail adding to table storage for the device simulator just continue
+                        Trace.TraceError("Failed to add enabled simulated device : {0}", ex.Message);
```
Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs]: maybe more specific logging? "Failed to add enabled device to simulated device storage. Device telemetry is expected not to be send." something like this ?

File: DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs [hunk 2]
```diff
@@ -349,6 +349,41 @@ await _virtualDeviceStorage.AddOrUpdateDeviceAsync(new InitialDeviceConfig()
...
+                {
+                    try
+                    {
+                        await _virtualDeviceStorage.RemoveDeviceAsync(deviceId);
+                    }
+                    catch (Exception ex)
+                    {
+                        //if an exception occurs while attempting to remove the
+                        //simulated device from table storage do not roll back the changes.
+                        Trace.TraceError("Failed to remove disabled simulated device : {0}", ex.Message);
```
Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs]: Failed to remove disabled device from simulated device store. Device will keep sending telemetry data. ?

File: DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs [hunk 3]
```diff
@@ -349,6 +349,41 @@ await _virtualDeviceStorage.AddOrUpdateDeviceAsync(new InitialDeviceConfig()
                 capturedException.Throw();
             }
 
+            if (repositoryDevice != null && repositoryDevice.IsSimulatedDevice)
```
Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs]: can you add some unit tests for this part? If you wish you can put this code in a public method called something like: AddOrRemoveSimulatedDevice() and test that method

**Pull request merged**